src/game_updater/GameUpdater.cpp

```diff
@@ -100,69 +136,76 @@ void GameUpdater::UpdatePlayer(std::shared_ptr<GameState> game_state) {
+                                             game_state->GetLevel()->getTree());
+  std::vector<std::shared_ptr<GameObject>> colliding_platforms;
+  for (std::shared_ptr<GameObject> g : *collidingObjects) {
+    if (g->GetType() == ObjectType::COLLECTIBLE) {
+      if (std::shared_ptr<Collectible> c =
+              std::static_pointer_cast<Collectible>(g)) {
+        if (!c->GetCollected()) {
+          c->SetCollected();
+          game_state->GetPlayer()->SetScore(
+              game_state->GetPlayer()->GetScore() + 1);
+        }
+      }
+    } else if (g->GetType() == ObjectType::OBSTACLE) {
+      colliding_platforms.push_back(g);
     }
   }
 
-  if (colliding_platforms.size() > 1) {
-    // TODO(jarhar): further consider what is happening when we are colliding
-    // with two objects
-    Reset(game_state);
-  } else if (colliding_platforms.size() == 0) {
-    // No collision
-  } else {
-    // Handle the collision
-    std::shared_ptr<GameObject> colliding_object = colliding_platforms[0];
+  // Handle the collisions
+  for (std::shared_ptr<GameObject> colliding_object : colliding_platforms) {
     // Are we in the air, and is the object we are colliding with below us?
     // If so, then set it as our current ground
-    if (!player->GetGround()) {
-      AxisAlignedBox colliding_box = colliding_object->GetBoundingBox();
-      AxisAlignedBox player_box = player->GetBoundingBox();
-
-      // If the player's bounding box was "above" the platform's box before
-      // collision,
-      // then we consider the collision "landing" on the platform
-      float prev_player_min_y = previous_player_box.GetMin().y;
-      float colliding_max_y = colliding_box.GetMax().y;
-      if (prev_player_min_y > colliding_max_y) {
-        // above colliding box, we are grounded on this platform
-        // now attach player to the ground
-        player->SetGround(colliding_object);
-        player->SetVerticalVelocity(0);
-        player->SetPosition(
-            glm::vec3(player->GetPosition().x,
-                      colliding_max_y + Player::PLATFORM_SPACING +
-                          (player_box.GetMax().y - player_box.GetMin().y) / 2,
-                      player->GetPosition().z));
-      } else {
-        // The collision was not from above, so reset the game
-        Reset(game_state);
-      }
+    AxisAlignedBox colliding_box = colliding_object->GetBoundingBox();
+    float colliding_max_y = colliding_object->GetBoundingBox().GetMax().y;
+    AxisAlignedBox player_box = player->GetBoundingBox();
+    float prev_player_min_y = previous_player_box.GetMin().y;
+
+    // If the player's bounding box was "above" (within 1 velocity unit +
+    // allowed COLLISION_WIDTH error) the platform's box before
+    // collision,
+    // then we consider the collision "landing" on the platform
+    if (std::abs(prev_player_min_y - colliding_max_y) <
+        (std::abs(player->GetVerticalVelocity()) + collision_width)) {
```

Comment:
I noticed that if the player falls for a long time onto a platform their bounding box moves below the colliding box before the game can detect a collision, this adds some tolerance for that based on the players vertical velocity.

tl;dr
it's so the player doesn't take falling damage